server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureState.java

```diff
@@ -31,6 +34,11 @@ public class SearchBackpressureState {
      */
     private final AtomicLong limitReachedCount = new AtomicLong();
 
+    /**
+     * Usage stats for the last cancelled task.
+     */
+    private final AtomicReference<CancelledTaskStats> lastCancelledTaskStats = new AtomicReference<>();
```

Comment:
@nssuresh2007 Can you add your thoughts here? Do you see any value in adding last N cancelled tasks, or maybe, even remove this for now completely?

Reply:
We thought of having last cancelled task stats to provide exact details of resource consumption when the task was cancelled. But since this information can be got from logs and is not much useful when lots of tasks get cancelled, I am fine with removing this section for now.

Reply:
Updated! Removed anything related to CancelledTaskStats for now.